File: DR/1_19/10828.cpp
#include <iostream>
#include <stack>
#include <string.h>

using namespace std;
int main() {
	char sen[50];
	int n, number;
	cin>>n;
	stack<int> s;
	
	for(int i = 0; i<n; i++) {
		cin>>sen;
		if(strcmp(sen, "push") == 0) {
			cin>>number;
			s.push(number);
		}
		else if(strcmp(sen, "top") == 0) {
			if(!s.empty()) {
				cout<<s.top()<<endl;
			}
			else {
				cout<<-1<<endl;
			}
		}
		else if(strcmp(sen, "size") == 0) {
			cout<<s.size()<<endl;
		}
		else if(strcmp(sen, "empty") == 0) {
			cout<<s.empty()<<endl;
		}
		else if(strcmp(sen, "pop") == 0) {
			if(!s.empty()) {
				cout<<s.top()<<endl;
				s.pop();
			}
			else {
				cout<<-1<<endl;
			}
		}
	}
}

File: EB/01.29/10828.py
# 10828. 스택

import sys
input = sys.stdin.readline

n = int(input())
stack = []

for _ in range(n):
    str = input().split()
    order = str[0]
    
    if order == "push":
        stack.append(str[1])

    elif order == "pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
            
    elif order == "size":
        print(len(stack))
        
    elif order == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)
            
    elif order == "top":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

'''
n = int(input())
stack = []

for i in range(n):

    str = input()
    
    if str.find("push") != -1:
        a, b = str.split()
        b = int(b)
        stack.append(b)

    elif str.find("pop") != -1:
        if len(stack) == 0:
            print("-1")
        else:
            print(stack.pop())
        
    elif str.find("size") != -1:
        print(len(stack))

    elif str.find("empty") != -1:
        if len(stack) == 0:
            print("1")
        else:
            print("0")
    elif str.find("top") != -1:
        if len(stack) == 0:
            print("-1")
        else:
            print(stack[len(stack)-1])

'''
